packages/flutter/test/widgets/automatic_keep_alive_test.dart

```diff
@@ -443,4 +443,50 @@ void main() {
     expect(find.byKey(const GlobalObjectKey<_LeafState>(5)), findsNothing);
     expect(find.byKey(const GlobalObjectKey<_LeafState>(0)), findsNothing);
   });
-}
\ No newline at end of file
+
+  testWidgets('AutomaticKeepAlive with keepAlive set to true before initState', (WidgetTester tester) async {
+    await tester.pumpWidget(new Directionality(
+      textDirection: TextDirection.ltr,
+      child: new ListView.builder(
+        itemCount: 50,
+        itemBuilder: (BuildContext context, int index){
+          if (index == 0){
+            return const _AlwaysKeepAlive(
+              key: const GlobalObjectKey<_AlwaysKeepAliveState>(0),
+            );
+          }
+          return new Container(
+            height: 44.0,
+            child: const Text('FooBar'),
+          );
+        },
+      ),
+    ));
+
+    expect(find.byKey(const GlobalObjectKey<_AlwaysKeepAliveState>(0)), findsOneWidget);
+    await tester.drag(find.byType(ListView), const Offset(0.0, -1000.0)); // move to bottom
+    await tester.pump();
+    expect(find.byKey(const GlobalObjectKey<_AlwaysKeepAliveState>(0)), findsOneWidget);
+  });
+}
+
+class _AlwaysKeepAlive extends StatefulWidget{
+  const _AlwaysKeepAlive({Key key}):super(key:key);
```

Comment:
trivial nits: space before `{` on the class line, spaces around first `:` and space after second `:` on the constructor line.